Contributing.md

```diff
@@ -30,10 +30,20 @@ This project uses clang-format (stable branch) to check for common style issues.
 
 In most cases, clang-format can and **should** be used to automatically reformat code and solve most formatting issues.
 
-- To run clang-format on all staged files:
+It can be downloaded as part of the LLVM package here: [LLVM Downloads](http://releases.llvm.org/download.html)
+
+- To run clang-format on all staged files use the following command:
   ```
   git diff --cached --name-only | egrep '[.](cpp|h|mm)$' | xargs clang-format -i
   ```
+  - **Warning:** This method does not take the `.clang-format` configuration file into account, results may vary. While it may perform good enough to pass Lint checks on github, it will not be 100% accurate.
```

Comment:
This is incorrect. clang-format (3.8, at least) defaults to -style=file, which means it will use a .clang-format file located in the parent directories of the source file.

Reply:
When I was testing I did so with 4 different versions, 3.8.0, 3.8.1, 3.8.9, 5.0.0 and none produced the same results as the one presumably used on upstream which @Stenzek said was some version below 3.8 or somewhere around there. The clang-format locally was fixing things without me even editing anything, I tried the recursive thing on the whole repo (which broke the repo so I had to reset or reclone not sure I frogot), but it was a bit time back so I'm not sure. I was also suspecting that the entire upstream isn't standardized to one clang-format behavior, but other people simply shut that idea down as being very minor to what I suggested, if indeed the settings file is being respected on most of these methods then it must be something else that's elusive or that it was isolated behavior on my part only.

Reply:
I forgot to say, I'm doing more testing now ...

Reply:
Looks like your right